packages/datagateway-common/src/api/datafiles.tsx

```diff
@@ -0,0 +1,372 @@
+  // need to use useDeepCompareEffect here because the array returned by useQueries
+  // is different every time this hook runs
+  useDeepCompareEffect(() => {
+    const currCountReturned = queries.reduce(
+      (acc, curr) => acc + (curr.isFetched ? 1 : 0),
+      0
+    );
+    const batchMax =
+      sizes.length - currCountReturned < 5
+        ? sizes.length - currCountReturned
+        : 5;
+    // this in effect batches our updates to only happen in batches >= 5
+    if (currCountReturned - countAppliedRef.current >= batchMax) {
+      setSizes(queries);
+      countAppliedRef.current = currCountReturned;
+    }
+  }, [queries]);
+
+  return sizes;
+};
+
+const fetchDatafileCount = (
+  apiUrl: string,
+  filters: FiltersType,
+  additionalFilters?: AdditionalFilters
+): Promise<number> => {
+  const params = getApiParams({ filters, sort: {} });
+  params.delete('order');
+
+  if (additionalFilters) {
+    additionalFilters.forEach((filter) => {
+      params.append(filter.filterType, filter.filterValue);
+    });
+  }
+
+  return axios
+    .get(`${apiUrl}/datafiles/count`, {
+      params,
+      headers: {
+        Authorization: `Bearer ${readSciGatewayToken().sessionId}`,
+      },
+    })
+    .then((response) => response.data);
+};
+
+export const useDatafileCount = (
+  additionalFilters?: AdditionalFilters
+): UseQueryResult<number, AxiosError> => {
+  const apiUrl = useSelector((state: StateType) => state.dgcommon.urls.apiUrl);
+  const location = useLocation();
+  const { filters } = parseSearchToQuery(location.search);
+
+  return useQuery<
+    number,
+    AxiosError,
+    number,
+    [string, string, { filters: FiltersType }, AdditionalFilters?]
+  >(
+    ['count', 'datafile', { filters }, additionalFilters],
+    (params) => {
+      const { filters } = params.queryKey[2];
+      return fetchDatafileCount(apiUrl, filters, additionalFilters);
+    },
+    {
+      placeholderData: 0,
+      onError: (error) => {
+        handleICATError(error);
+      },
+    }
+  );
+};
+
+const fetchDatafileDetails = (
+  apiUrl: string,
+  datafileId: number
+): Promise<Datafile> => {
+  const params = new URLSearchParams();
+  params.append('where', JSON.stringify({ id: { eq: datafileId } }));
+
+  return axios
+    .get(`${apiUrl}/datafiles`, {
+      params,
+      headers: {
+        Authorization: `Bearer ${readSciGatewayToken().sessionId}`,
+      },
+    })
+    .then((response) => response.data[0]);
+};
+
+export const useDatafileDetails = (
+  datafileId: number
+): UseQueryResult<Datafile, AxiosError> => {
+  const apiUrl = useSelector((state: StateType) => state.dgcommon.urls.apiUrl);
+
+  return useQuery<Datafile, AxiosError, Datafile, [string, number]>(
+    ['datafileDetails', datafileId],
+    (params) => fetchDatafileDetails(apiUrl, params.queryKey[1]),
+    {
+      onError: (error) => {
+        handleICATError(error);
+      },
+    }
+  );
+};
+
+export const downloadDatafileQuery = (
```

Comment:
Another note for later: change this to `downloadDatafile` once the `downloadDatafile` function in `src/state/actions/datafiles.tsx` has been totally removed from use